### ISSUE TITLE
generate the annotations for any dataset 

### ISSUE THREAD

**Issue by ChristineBasta** [artursstaf/mitigating-gender-bias-wmt-2020#3]
Thanks for sharing this interesting work; I just want to make sure of the steps I should follow to create the annotations successfully. First I call generate_genders for the gendered_language_file (i.e. train.de) 
and then I use align_genders to align it with the English file. I still do not get what 'randomly_include_genders.py' is actually doing. For a new language, I see that I need to add its packages and make all the steps of the bash scripts in each language mentioned. Is this right?
Thanks a lot!

**Comment by artursstaf** [comment]
Hi, sorry for late answer :(. I have just now updated README and latest scripts, please get latest revision. 
Regarding `randomly_include_genders.py` this script applies random dropout to gender annotations described in paper.
I also tried to give some context to other key scripts in README. 
Regarding experiments for new language pair - yes reproducing steps in bash scripts in `scripts/{language}` should be sufficient (I tried to make it as general as possible). 
`scripts/common/genders2_pre-process.sh` describes data pre-processing steps necessary to get TGA training data.

1.  Extract gender annotations from target sentences `generate_genders.py`
2.  Generate word alignments with `fast_align`
3.  Align gender annotations generated from target words to source words `align_genders.py`
4.  Apply annotation dropout `randomly_include_genders.py`
5.  Word level gender annotations are copied to subword format `genders_bpe.py `

Some scripts were added in a hurry and I didn't have time to test them in end-end scenario. If you encounter any issues with reproducing experiments or applying TGA to a new language please contact me, I will be happy to help.

**Comment by ChristineBasta** [comment]
Thanks for the detailed answer!I appreciate it!
